premailer/tests/test_premailer.py

```diff
@@ -2984,3 +2984,57 @@ def test_pseudo_selectors_without_selector(self):
         p = Premailer(html, exclude_pseudoclasses=False, keep_style_tags=True)
         result_html = p.transform()
         compare_html(expect_html, result_html)
+
+    def test_preserve_handlebar_syntax_false(self):
+        """Demonstrate encoding of handlebar syntax with preservation disabled.
+
+        Original issue: https://github.com/peterbe/premailer/issues/248
+        """
+
+        html = """
+            <html>
+            <img src="{{ data | default: 'Test & <code>' }}">
+            <a href="{{ data | default: "Test & <code>" }}"></a>
+            </html>
+        """
+
+        expect_html = """
+<html>
+    <head>
+    </head>
+    <body>
+    <img src="%7B%7B%20data%20%7C%20default:%20'Test%20&amp;%20&lt;code&gt;'%20%7D%7D">
+    <a href="%7B%7B%20data%20%7C%20default:%20" test>" }}"&gt;</a>
+    </body>
+</html>
+        """
+        p = Premailer(html)
+        result_html = p.transform()
+        compare_html(expect_html, result_html)
+
+    def test_preserve_handlebar_syntax_true(self):
```

Comment:
Can you please collapse these two tests into one. The first one is kinda crazy in that it doesn't actually test anything. It just does what it would do if you don't apply the functionality :)
But I quite like the effect that it demonstrates, so I think it's worth keeping. 

If you just had _1_ test for both `preserve_handlebar_syntax=True` and `preserve_handlebar_syntax=False` you wouldn't need to repeat the input string. And you wouldn't need to repeat the comment about which original issue this revolves around.

Reply:
Sure. Here is the commit where I combined the two tests: https://github.com/peterbe/premailer/pull/252/commits/0dc619286a594bab7a1f716d74c094542561718a

As a side-note, in defense of having two tests:
- The first test, with the setting turned off, tests a case in which using handlebar syntax breaks the encoding. While I'm not sure it warranted its own test, I felt that it was good to show to the user (hence including in the 1, combined test). And, because I thought it warranted being included in a test, I followed the ["Arrange Act Assert"](http://wiki.c2.com/?ArrangeActAssert) test pattern.
   - Combining the two tests breaks that testing pattern because it requires the test to "Act" twice.

All that to say, I'm totally cool with combining them, I just wanted to say why I had two to begin with :)